pkg/gardener/shoot/extender/extensions/dns.go

```diff
@@ -74,9 +73,21 @@ func newDNSExtensionConfig(domain, secretName, dnsProviderType string) *DNSExten
 }
 
 func NewDNSExtension(shootName, secretName, domainSuffix, dnsProviderType string) (*gardener.Extension, error) {
-	domain := fmt.Sprintf("%s.%s", shootName, domainSuffix)
+	var providerConfig *DNSExtensionProviderConfig
+
+	if secretName != "" && dnsProviderType != "" {
+		domain := fmt.Sprintf("%s.%s", shootName, domainSuffix)
+		providerConfig = newDNSExtensionConfig(domain, secretName, dnsProviderType)
+	} else {
+		// special case for Gardener's DNS solution
+		providerConfig = &DNSExtensionProviderConfig{
```

Comment:
could you add a unit test that checks if this extension data will be generated correctly if `secretName` or `dnsProviderType` are empty?